packages/thirdweb/src/react/core/hooks/connection/useAutoConnect.tsx

```diff
@@ -193,6 +206,9 @@ export function AutoConnect(props: AutoConnectProps) {
           });
 
           connect(activeWallet);
+          if (onConnect) {
```

Comment:
Should `onConnect` be called outside the try-catch so the user can provide their own error handling?

Reply:
error will be thrown before the wallet is connected, so I'm not sure how would `onConnect` handle the error here?

Reply:
Just thinking if the user passes a function that errors to `onConnect` and doesn't handle it, the `connect` try-catch will be triggered and they'll get an error "Failed to auto connect last active wallet" which misleads them to think `connect` failed instead of `onConnect`

Reply:
ok - let me wrap it in try-catch

Reply:
done